torch/onnx/symbolic_helper.py

```diff
@@ -177,6 +177,17 @@ def _is_tensor(x):
 def _is_tensor_list(x):
     return isinstance(x.type(), torch._C.ListType) and isinstance(x.type().getElementType(), torch._C.TensorType)
 
+def _get_tensor_rank(x):
+    if not _is_tensor(x) or x.type() is None:
+        return None
+    return x.type().dim()
+
+def _get_tensor_sizes(x, allow_symbol=True):
+    if not _is_tensor(x) or x.type() is None:
+        return None
+    if allow_symbol:
+        return x.type().varyingSizes()
+    return x.type().sizes()
```

Comment:
If `allow_symbol`, the size contains symbols? If not, then the dynamic size would return `None` for that dimension?

Reply:
added comments for details.